fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/api/FedoraToOCFLObjectIndex.java

```diff
@@ -40,5 +40,14 @@
      * @return the mapping
      */
     FedoraOCFLMapping getMapping(final String fedoraResourceIdentifier);
+
+    /**
+     * Adds a mapping to the index
+     * @param fedoraResourceIdentifier The fedora resource
+     * @param parentFedoraResourceId The parent of the fedora resource
+     * @param ocflObjectId The ocfl object id
+     */
+    public void addMapping(final String fedoraResourceIdentifier, final String parentFedoraResourceId,
+                           final String ocflObjectId);
```

Comment:
How do we add a mapping for the root of an OCFL object? Is the `fedoraResourceIdentifier == parentFedoraResourceId`?

Reply:
yes

Reply:
Could you add a test to verify the root resource case?

Also, I'm not sure at the moment where the root's parent gets set to itself? I wonder if it'd make sense for the parent of the root to be null instead, otherwise it might come back as its own child in some queries, unless i'm misunderstanding.

Reply:
I think the confusion may be around terminology. It's not really the "parent" it is the fedora resource that corresponds directly to the OCFL object. I'm not sure what to call that.